PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,104 @@
+        private readonly IPageDialogService _pageDialogService;
+        public ICommand GoToApiContactsPageCommand { get; set; }
+        public ICommand GoToHomePageCommand { get; set; }
+        public ICommand ChangeLanguageCommand { get; set; }
+        public ICommand TurnOnFlashLightCommand { get; set; }
+        public ICommand TurnOffFlashLightCommand { get; set; }
+        private string _title;
+        private MyLanguage _selectedLanguage;
+        private ObservableCollection<MyLanguage> _supportedLanguage;
+        public string Title
+        {
+            get
+            {
+                return _title;
+            }
+            set
+            {
+                SetProperty(ref _title, value);
+            }
+        }
+        public ObservableCollection<MyLanguage> SupportedLanguage
+        {
+            get { return _supportedLanguage; }
+            set { SetProperty(ref _supportedLanguage, value); }
+        }
+        public MyLanguage SelectedLanguage
+        {
+            get { return _selectedLanguage; }
+            set { SetProperty(ref _selectedLanguage, value); }
+        }
+        public MainPageViewModel(INavigationService navigationService, IPageDialogService pageDialogService)
+        {
+            Title = "Main Page";
+            _navigationService = navigationService;
+            _pageDialogService = pageDialogService;
+            GoToApiContactsPageCommand = new DelegateCommand(GoToApiContactsPageCommandHandler);
+            GoToHomePageCommand = new DelegateCommand(GoToHomePageCommandHandler);   
+            ChangeLanguageCommand = new Command(PerformOperation);
+            TurnOnFlashLightCommand = new Command(TurnOnFlashLightCommandHandler);
+            TurnOffFlashLightCommand = new Command(TurnOffFlashLightCommandHandler);
+            SupportedLanguage = new ObservableCollection<MyLanguage>()
+            {
+                new MyLanguage{Name=AppResource.English,CI="en"},
+                new MyLanguage{Name=AppResource.French,CI="fr"},
+                new MyLanguage{Name=AppResource.Tamil,CI="ta"}
+            };
+            SelectedLanguage = SupportedLanguage.FirstOrDefault(x => x.CI == LocalizationResourceManager.Current.CurrentCulture.TwoLetterISOLanguageName);
+        }
+        private void PerformOperation(object obj)
+        {
+            LocalizationResourceManager.Current.SetCulture(CultureInfo.GetCultureInfo(SelectedLanguage.CI));
+        }
+        private async void GoToApiContactsPageCommandHandler()
+        {
+            await _navigationService.NavigateAsync("ApiContactsPage");
+        }
+        private async void GoToHomePageCommandHandler()
+        {
+            await _navigationService.NavigateAsync("HomePage");
+        }
+        public Task<bool> CanNavigateAsync(INavigationParameters parameters)
+        {
+            return _pageDialogService.DisplayAlertAsync(Title, "Navigate", "Yes", "No");
+        }
```

Comment:
Move all these to texts to resource file. Through out the app

Reply:
moved all the texts to resx file